index.js

```diff
@@ -161,11 +168,18 @@ function findNodeAtOffset(node, offset, predicate, parentNodes) {
 }
 
 // See https://www.ecma-international.org/ecma-262/5.1/#sec-A.5
-function isSourceElement(node) {
+function isSourceElement(opts, node) {
   if (node == null) {
     return false;
   }
   switch (node.type) {
+    case "ObjectExpression": // JSON
+    case "ArrayExpression": // JSON
+    case "StringLiteral": // JSON
+    case "NumericLiteral": // JSON
+    case "BooleanLiteral": // JSON
+    case "NullLiteral": // JSON
+      return opts.parser === "json";
```

Comment:
No, there's also [`ObjectProperty` and `json-identifier`](https://github.com/josephfrazier/prettier/blob/29631046cddbfcd5cb38b94cc5b333be51fccb60/src/parser-json.js#L30-L31). I guess the identifier is technically formattable, so maybe it should be included too.